src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java

```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
 
     @Override
     public boolean removeAll(Collection<?> c) {
```

Comment:
This is actually a breaking change.    The method contract is that "true" is returned whenever the class is modified.  I agree its weird but that's from the Java Language, we are not free to alter it here.      Btw, doesn't your implementation always return false?    You might want to revisit that logic as well.